### PR TITLE
feat: library-go bump

### PR DESCRIPTION
**Pull request by eggfoobar**
Bumping the library-go dependency to latest, this should take advantage of leader election changes for SNO clusters proposed in this [library-go PR](https://github.com/openshift/library-go/pull/1242) and performance improvements in this [PR](https://github.com/openshift/library-go/pull/1235).

Changes:
  - updated library-go to latest

Signed-off-by: ehila <ehila@redhat.com>

### CODE REVIEW

**Review by bertinatto — COMMENTED** [review]
Our CSI sidexar containers also use the default leader election values provided in library-go:

https://github.com/openshift/library-go/blob/ed9bc958bd8a2fff079d52976806e4e0a8a7c315/pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go#L75

Do we need to update those as well?

File: pkg/operator/csidriveroperator/crcontroller.go [hunk 1]
```diff
@@ -222,7 +222,7 @@ func (c *CSIDriverOperatorCRController) syncConditions(conditions []operatorapi.
 		degradedCnd.Status = operatorapi.ConditionFalse
 	}
 
-	_, _, err := v1helpers.UpdateStatus(c.operatorClient,
+	_, _, err := v1helpers.UpdateStatus(context.TODO(), c.operatorClient,
```
[the remark this answers is not in the document]
Comment (eggfoobar) [reply pkg/operator/csidriveroperator/crcontroller.go]: Absolutely, updated with recommendations.

File: pkg/operator/csidriveroperator/olmremovalcontroller.go [hunk 2]
```diff
@@ -317,7 +317,7 @@ func (c *OLMOperatorRemovalController) markFinished(message string) error {
 	if err := c.saveMetadata("", ""); err != nil {
 		return err
 	}
-	_, _, err := v1helpers.UpdateStatus(c.operatorClient,
+	_, _, err := v1helpers.UpdateStatus(context.TODO(), c.operatorClient,
```
[the remark this answers is not in the document]
Comment (eggfoobar) [reply pkg/operator/csidriveroperator/olmremovalcontroller.go]: done

File: pkg/operator/csidriveroperator/crcontroller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (eggfoobar) [reply pkg/operator/csidriveroperator/crcontroller.go]: I did go ahead and modify other places I was using TODO to grab from the parent as well, if that's not something you're looking for I can revert it back

File: cmd/cluster-storage-operator/main.go [hunk 3]
```diff
@@ -22,6 +22,7 @@ func main() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 
 	logs.InitLogs()
+	logs.AddFlags(pflag.CommandLine)
```
Comment (bertinatto) [inline cmd/cluster-storage-operator/main.go:25]: Please check this PR for an example of how this should look like: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/581/commits/c339a947622142be4e6e0df7799cb2a1445e1eb2

Comment (eggfoobar) [reply cmd/cluster-storage-operator/main.go]: Oh that's way better, thank you, updated :)

**Review by bertinatto — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**